lib/mumukit/platform/application.rb

```diff
@@ -27,16 +27,10 @@ def url_for(path)
     uri.url_for(path) if path
   end
 
+  # warning: this code is tightly coupled to the fact that
+  # applications can only rebuild urls in fragmented-mode
   def organic_url_for(organization, path)
-    uri = organic_uri(organization)
-    # warning: this code is tightly
-    # coupled to the fact that applications can only rebuild urls
-    # in fragmented-mode
-    if uri.fragment
-      uri.to_s + relative_path(path)
-    else
-      uri.url_for(relative_path path)
-    end
+    organic_uri(organization).url_for(relative_path path)
```

Comment:
The problem is that this code is quite hacky but makes sense in application - where such assumptions are always met - but in the URI class.

Reply:
However, we could rename the missleading `url_for` method and/or add a `fragmented` optional flag. That way moving it to `URI` would make more sense. 

It could be called `append_path` for example.